File: BE/hieuthuoc/src/main/java/com/example/hieuthuoc/entity/DanhGia.java
package com.example.hieuthuoc.entity;

import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@Entity
@Table(name = "danh_gia")
@EntityListeners(AuditingEntityListener.class)
@EqualsAndHashCode(callSuper = true)
public class DanhGia extends TimeAuditable{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "thuoc_id")
    private Thuoc thuoc;

    @ManyToOne
    @JoinColumn(name = "nguoi_dung_id")
    private NguoiDung nguoiDung;
    
    @ManyToOne
    @JoinColumn(name = "danh_gia_id")
    private DanhGia danhGiaGoc;

    private Integer diemSo;
    
	@Column(columnDefinition = "TEXT")
    private String danhGia;

    // getters and setters
}

File: front/src/app/_model/danhgia.ts
import { NguoiDung } from "./auth/nguoidung";

export class DanhGia {
  id?: string;
  dangGiaGocId?: string;
  nguoiDungId?: string;
  thuocId?: string;
  createdAt?: Date;
  updatedAt?: Date;
  danhGia?: string;
  diemSo?: number;
  ngayDanhGia?: number;

  nguoiDung?: NguoiDung;
}
